pkg/routeagent_driver/handlers/ovn/gateway_route_handler.go

```diff
@@ -124,6 +129,16 @@ func (h *GatewayRouteHandler) TransitionToGateway() error {
 
 	h.isGateway = true
 
+	for _, endpoint := range h.remoteEndpoints {
+		// This piece of code is designed to manage upgrades from a version lower than 0.16.3 to a higher version,
+		// where we utilize the endpoint name as the identifier for gwr. It can be removed once we stop supporting
+		// the 0.16 version.
+		if err := h.smClient.SubmarinerV1().GatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+			endpoint.Name, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "error deleting gatewayRoute %q", endpoint.Name)
+		}
```

Comment:
This can be combined with the loop below on L142.

Reply:
Rather than introducing new changes in this backport PR, I feel @aswinsuryan can push a separate PR on master and backport it (once its merged).

Reply:
The **devel** version does not have 2 loops.

Reply:
aah aa.. I thought this was a clean-cherry pick, its not. It makes sense then.